src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ProjectTests.kt

```diff
@@ -282,8 +282,48 @@ class ProjectTests : VersionFileTests()
     }
 
     @Test
-    fun `can return expected English error when delete nonexistent version`()
-    {
+    fun `can promote version`() {
+        val createResult = createProject()
+        val createProjectData = getResponseData(createResult)
+        val projectId = createProjectData["id"].asInt()
+        val versions = createProjectData["versions"] as ArrayNode
+        val versionId = versions[0]["id"].asText()
+        getUpdateVersionStateResult(projectId, versionId, testState)
+
+
+        val map = LinkedMultiValueMap<String, String>()
+        map.add("name", "newProject")
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_FORM_URLENCODED
+        val httpEntity = HttpEntity(map, headers)
+
+        testRestTemplate.postForEntity<String>("/project/$projectId/version/$versionId/promote", httpEntity)
+
+        val result = testRestTemplate.getForEntity<String>("/projects/")
+        val data = getResponseData(result) as ArrayNode
+        assertThat(data.count()).isEqualTo(2)
+        val versionsData = data[0]["versions"] as ArrayNode
```

Comment:
so versionsData is the versions belonging to your new project, not the old one? Should be, because that's been updated most recently, I think. Maybe worth checking the project name just to be sure.